### PR TITLE
hiera integration

### PR DESCRIPTION
**Pull request by clemoule**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -52,3 +52,6 @@ RUN set -eux && groupadd --gid $USER_GUID $GROUPNAME \
 
 WORKDIR $HOME_DIR
 RUN curl -fsSL https://code-server.dev/install.sh | sh
+RUN pip install timm
+RUN pip install timm --upgrade
+RUN pip install tqdm
```
Comment (LBerth) [inline Dockerfile:57]: put in requirements.txt

File: py4cast/models/vision/unetrpp.py [hunk 2]
```diff
@@ -434,6 +435,7 @@ def _init_weights(self, m):
     def forward_features(self, x):
         hidden_states = []
 
+        # UTILE ?
```
Comment (LBerth) [inline py4cast/models/vision/unetrpp.py:438]: remove comments `utile ?`

File: requirements.txt [hunk 3]
```diff
@@ -25,6 +25,9 @@ onnxscript==0.1.0.dev20241018
 monai==1.4.0
 gif==23.3.0
 scikit-image==0.24.0
+typer==0.9.0
+tensorflow==2.13.1
+py-spy
```
Comment (LBerth) [inline requirements.txt:29]: add version of py-spy

File: tests/test_models.py
Comment (LBerth) [inline tests/test_models.py:1]: I don't think you need to modify the tests, except to add hiera in the list of models

File: py4cast/models/vision/hiera.py [hunk 4]
```diff
@@ -0,0 +1,1412 @@
...
+            )
+        # print("x in hiera forward after mask is not none", x.shape)
+        skip_list = []
+        for i, blk in enumerate(self.blocks):
+            if i in self.stage_ends:
+                skip_list.append(x)
+            x = blk(x)
+            if return_intermediates and i in self.stage_ends:
+                intermediates.append(self.reroll(x, i, mask=mask))
+            # print("x in hiera forward after blk=",i, x.shape)
```
Comment (LBerth) [inline py4cast/models/vision/hiera.py:544]: remove commented prints

File: py4cast/models/vision/hiera.py [hunk 5]
```diff
@@ -0,0 +1,1412 @@
...
+    ) -> torch.Tensor:
+        x = do_masked_conv(x, self.proj, mask)
+        x = x.reshape(x.shape[0], x.shape[1], -1).transpose(2, 1)
+        return x
+
+
+# ------------------------------------------------------------------------------#
+
+class _a:
+    a=1
```
Comment (LBerth) [inline py4cast/models/vision/hiera.py:1002]: remove this

File: py4cast/models/vision/hiera.py [hunk 6]
```diff
@@ -0,0 +1,1412 @@
...
+        B, _, C = x.shape
+        x = x.reshape(B, self.input_shape[0], self.input_shape[1], C)
+        return x
+
+
+# ------------------------------------------------------------------------------#
+
+class _b:
+    a=1
+
```
Comment (LBerth) [inline py4cast/models/vision/hiera.py:1102]: rm

File: py4cast/models/vision/hiera.py [hunk 7]
```diff
@@ -0,0 +1,1412 @@
...
+                    ]
+                )
+            )
+        return layers
+
+
+# ------------------------------------------------------------------------------#
+class _c:
+    a=1
+
```
Comment (LBerth) [inline py4cast/models/vision/hiera.py:1306]: rm

File: py4cast/MAE_lightning.py [hunk 8]
```diff
@@ -0,0 +1,366 @@
...
+#             config_override: Union[Dict, None] = None,
+#             batch_size: int = 1,
+#             num_workers: int = 1,
+#             pin_memory: bool = False,
+#             prefetch_factor: Union[int, None] = None,
+#             persistent_workers: bool = False,
+#             num_input_steps: int = 1,
+#             num_pred_steps_train: int = 1,
+#             num_pred_steps_val_test: int = 1,
+#     ):
```
Comment (LBerth) [inline py4cast/MAE_lightning.py:94]: rm commented code

File: py4cast/MAE_lightning_no_cli
Comment (LBerth) [inline py4cast/MAE_lightning_no_cli:1]: this file misses the `.py` extension

File: py4cast/datasets/base.py [hunk 9]
```diff
@@ -705,6 +704,19 @@ def torch_dataloader(self, tl_settings: TorchDataloaderSettings) -> DataLoader:
...
+    # @abstractmethod
+    # def torch_dataloader_bis(
+    #     self,
+    #     batch_size: int = 1,
+    #     num_workers: int = 1,
+    #     pin_memory: bool = False,
+    #     prefetch_factor: Union[int, None] = None,
+    #     persistent_workers: bool = False,
+    # ) -> DataLoader:
+    #     """
```
Comment (LBerth) [inline py4cast/datasets/base.py:716]: can it be removed ?

File: py4cast/datasets/poesy.py
Comment (LBerth) [inline py4cast/datasets/poesy.py:1]: can you undo the modifications to poesy and smeago  ?

File: requirements.txt [hunk 10]
```diff
@@ -25,6 +25,15 @@ onnxscript==0.1.0.dev20241018
 monai==1.4.0
 gif==23.3.0
 scikit-image==0.24.0
+py-spy==0.4.0
```
Comment (A669015) [inline requirements.txt:28]: Could you please check if the packages added are really required ? I doesn't found any import of some of them in the comited code.

File: py4cast/MAE_lightning.py [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+                tb.add_image(
+                    f"val_plots/test_image_field_{i}", field_image, dataformats="HW"
+                )
+            # tb.add_image("val_plots/test_image", y_hat.permute(0, 3, 1, 2)[0], step, dataformats=dformat)
+
+    def validation_step(self, batch, batch_idx):
+        x, y = (
+            torch.squeeze(batch.inputs.tensor[:, 0, :, :, :], dim=1),
+            torch.squeeze(batch.outputs.tensor[:, 0, :, :, :], dim=1),
+        )
```
Comment (LBerth) [inline py4cast/MAE_lightning.py:237]: can it be put in `shared_forward_step` ?

File: requirements.txt [hunk 12]
```diff
@@ -25,6 +25,15 @@ onnxscript==0.1.0.dev20241018
...
 typer==0.12.5
 transformers==4.45.2
 torch-geometric==2.6.1
+timm==1.0.11
+tqdm==4.67.0 
+evaluate==0.4.3
+h5netcdf==1.4.0
+wget==3.2
+pyyaml==6.0.2
+torch==2.5.1
```
Comment (A669015) [inline requirements.txt:38]: Please do not add torch to the `requirements.txt` file, its version is managed in the `env.yaml` file (and in the `Dockerfile`).

**Pull request closed without merging**